src/commands/createWebApp/setPostPromptDefaults.ts

```diff
@@ -27,21 +28,27 @@ export async function setPostPromptDefaults(wizardContext: IWebAppWizardContext,
     await LocationListStep.setLocation(wizardContext, config.location || nonNullProp(defaultLocation, 'name'));
     const location: SubscriptionModels.Location = nonNullProp(wizardContext, 'location');
 
-    const defaultName: string = `appsvc_${wizardContext.newSiteOS}_${location.name}`;
+    let defaultName: string = `appsvc_${wizardContext.newSiteOS}_${location.name}`;
+    const newSkuTier = nonNullProp(nonNullProp(wizardContext, 'newPlanSku'), 'tier').toLowerCase();
+    if (newSkuTier !== freeTier) {
+        // Use "premium" instead of "premium v2"
+        const simpleTierName = newSkuTier.split(/v[0-9]/)[0].trim();
+        defaultName += `_${simpleTierName}`;
+    }
     const defaultGroupName: string = config.group || defaultName;
     const defaultPlanName: string = defaultName;
 
     const client: WebSiteManagementClient = await createWebSiteClient(wizardContext);
     const asp: WebSiteManagementModels.AppServicePlan | undefined = await tryGetAppServicePlan(client, defaultGroupName, defaultPlanName);
-    if (asp && checkPlanForPerformanceDrop(asp)) {
+    const hasPerfDrop = checkPlanForPerformanceDrop(asp);
+    if (asp && (hasPerfDrop || !matchesTier(asp, newSkuTier))) {
         // Subscriptions can only have 1 free tier Linux plan so show a warning if there are too many apps on the plan
-        if (wizardContext.newSiteOS === WebsiteOS.linux) {
+        if (wizardContext.newSiteOS === WebsiteOS.linux && newSkuTier === freeTier && hasPerfDrop) {
```

Comment:
Shouldn't the `newSkuTier === basicTier` as well?  Though I'm not sure why you need to check `newSkuTier === freeTier` here because `hasPerfDrop` should have covered that already.

Reply:
See the comment right above this line (which I'm pretty sure you wrote?):
> // Subscriptions can only have 1 free tier Linux plan so show a warning if there are too many apps on the plan

This is a unique case because linux free only supports 1 plan per subscription. Users can have multiple linux basic plans, so we want to go to the else case for linux basic.